static/src/javascripts/bootstraps/enhanced/liveblog.js

```diff
@@ -1,100 +1,164 @@
+import bean from 'bean';
+import { scrollToElement } from 'lib/scroller';
+import {
+	submitClickEvent,
+	submitInsertEvent,
+	submitViewEvent,
+} from 'common/modules/commercial/acquisitions-ophan';
+import { elementInView } from 'lib/element-inview';
 
 const affixTimeline = () => {
-    const keywordIds = config.get('page.keywordIds', '');
-
-    if (
-        isBreakpoint({
-            min: 'desktop',
-        }) &&
-        !keywordIds.includes('football/football') &&
-        !keywordIds.includes('sport/rugby-union')
-    ) {
-        // eslint-disable-next-line no-new
-        new Affix({
-            element: document.querySelector(
-                '.js-live-blog__sticky-components-container'
-            ),
-            topMarker: document.querySelector('.js-top-marker'),
-            bottomMarker: document.querySelector('.js-bottom-marker'),
-            containerElement: document.querySelector(
-                '.js-live-blog__sticky-components'
-            ),
-        });
-    }
+	const keywordIds = config.get('page.keywordIds', '');
+
+	if (
+		isBreakpoint({
+			min: 'desktop',
+		}) &&
+		!keywordIds.includes('football/football') &&
+		!keywordIds.includes('sport/rugby-union')
+	) {
+		// eslint-disable-next-line no-new
+		new Affix({
+			element: document.querySelector(
+				'.js-live-blog__sticky-components-container',
+			),
+			topMarker: document.querySelector('.js-top-marker'),
+			bottomMarker: document.querySelector('.js-bottom-marker'),
+			containerElement: document.querySelector(
+				'.js-live-blog__sticky-components',
+			),
+		});
+	}
 };
 
 const initFilterCheckbox = () => {
-    const filterKeyEvents = window.location.search.includes('filterKeyEvents=true');
-    const filterSwitch = document.getElementById("filter-switch");
+	const filterKeyEvents = window.location.search.includes(
+		'filterKeyEvents=true',
+	);
+	const filterSwitch = document.getElementById('filter-switch');
 
-    if (filterSwitch) {
-        filterSwitch.checked = filterKeyEvents;
-    }
-}
+	if (filterSwitch) {
+		filterSwitch.checked = filterKeyEvents;
+	}
+};
 
 const createAutoUpdate = () => {
-    if (config.get('page.isLive')) {
-        autoUpdate();
-    }
+	if (config.get('page.isLive')) {
+		autoUpdate();
+	}
 };
 
 const keepTimestampsCurrent = () => {
-    window.setInterval(() => initRelativeDates(), 60000);
+	window.setInterval(() => initRelativeDates(), 60000);
 };
 
 const isVisible = (element) => {
-    const position = element && element.getBoundingClientRect();
-    return position && position.top >= 0 && position.bottom <= window.innerHeight
-}
+	const position = element && element.getBoundingClientRect();
+	return (
+		position && position.top >= 0 && position.bottom <= window.innerHeight
+	);
+};
+
+const cleanId = (blockIdString) => {
+	return blockIdString.substring(0, blockIdString.indexOf('-pinned'));
+};
+
+const buildComponentEventWithoutAction = (pinnedBlockId) => ({
```

Comment:
I think it might make the code a bit simpler if this object were inlined in the calls to `sendInsertEvent` and `sendViewEvent`. There'd be fewer layers of abstraction to reason about

Reply:
Fair point, I'll refactor to remove the abstraction